tools/include/edm4hep2json.hxx

```diff
@@ -91,28 +105,207 @@ nlohmann::json processEvent(const podio::Frame& frame, std::vector<std::string>&
       nlohmann::json jsonColl{
           {collList[i], {{"collection", hitCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
       jsonDict.insert(jsonColl.begin(), jsonColl.end());
-    } else if (coll->getTypeName() == "edm4hep::SimCalorimeterHitCollection") {
-      auto& hitCollection = frame.get<edm4hep::SimCalorimeterHitCollection>(collList[i]);
+    } else if (coll->getTypeName() == "edm4hep::ParticleIDCollection") {
+      auto& particleIDCollection = frame.get<edm4hep::ParticleIDCollection>(collList[i]);
       nlohmann::json jsonColl{
-          {collList[i], {{"collection", hitCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
+          {collList[i],
+           {{"collection", particleIDCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
       jsonDict.insert(jsonColl.begin(), jsonColl.end());
     } else if (coll->getTypeName() == "edm4hep::ClusterCollection") {
       auto& clusterCollection = frame.get<edm4hep::ClusterCollection>(collList[i]);
       nlohmann::json jsonColl{
           {collList[i],
            {{"collection", clusterCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
       jsonDict.insert(jsonColl.begin(), jsonColl.end());
+    } else if (coll->getTypeName() == "edm4hep::TrackerHitCollection") {
+      auto& hitCollection = frame.get<edm4hep::TrackerHitCollection>(collList[i]);
+      nlohmann::json jsonColl{
+          {collList[i], {{"collection", hitCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
+      jsonDict.insert(jsonColl.begin(), jsonColl.end());
+    } else if (coll->getTypeName() == "edm4hep::TrackerHitPlaneCollection") {
+      auto& hitPlaneCollection = frame.get<edm4hep::TrackerHitPlaneCollection>(collList[i]);
+      nlohmann::json jsonColl{
+          {collList[i],
+           {{"collection", hitPlaneCollection}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
+      jsonDict.insert(jsonColl.begin(), jsonColl.end());
```

Comment:
Could this be made into a function template that is then simply called here for every case? That would improve readability quite a bit, I think, something like the following should work:

```cpp
template<typename CollT>
void insertIntoJson(nlohmann::json& jsonDict, const podio::CollectionBase* coll, const std::string& name) {
  const auto* typedColl = static_cast<const CollT*>(coll); // safe to cast, since we have queried the type before
  nlohmann::json jsonColl{
          {name,
           {{"collection", *typedColl}, {"collID", coll->getID()}, {"collType", coll->getTypeName()}}}};
  jsonDict.insert(jsonColl.begin(), jsonColl.end());
}
```
and then making the `if`-`else` chain something like
```cpp
if (coll->getTypeName() == "edm4hep::TrackCollection") {
  insertIntoJson<edm4hep::TrackCollection>(jsonDict, coll, collList[i]);
} else if (/*...*/){
  // ...
}
```

Reply:
Yeah, this is much easier to read :) Done